### DISCUSSION TITLE
Call for Feedback: What’s Next for WooCommerce Tax?

### DISCUSSION THREAD

**Discussion by brentmackinnon** [woocommerce/woocommerce#52463] [Request for Comments]
## Improving Tax Management in WooCommerce: Community Feedback

We're evaluating ways to enhance tax handling in WooCommerce and need your input to ensure we're building solutions that address real-world needs.

#### Current Challenges

- Complex tax calculations across multiple jurisdictions
- Manual updates required for changing tax regulations
- Limited automation options for global commerce
- Varying needs between small and enterprise merchants

#### Areas We're Exploring

- Automated tax calculations
- Enhanced compliance tools
- Improved reporting capabilities
- Better integration with tax services
- Custom tax rule management

#### Technical Considerations

- Integration with existing tax calculation flows
- Performance impact of automated calculations
- Data storage for complex tax rules
- API endpoints for custom implementations
- Compatibility with existing tax plugins

### What's your current tax management setup?

- Custom code implementations
- Third-party plugins used
- Manual processes

### What are your most challenging tax scenarios?

- Multiple tax jurisdictions
- Special tax rules
- Reporting requirements
- etc. 

### Technical preferences:

- API requirements
- Performance considerations
- Integration needs
- etc. 

### Impact on Existing Implementations

We want to ensure any improvements maintain compatibility with existing setups while adding new capabilities. Developers currently using custom tax implementations should be able to transition smoothly to any new features.

Let us know what you think!

**Comment by shahbazaldiablo** [comment]
Hi @brentmackinnon,
You guys are already doing awesome work...

**Comment by tommiby** [comment]
Hello,
thanks for amazing job!
We are using REST API to fetch order data from several WooCommerce stores. 
Could you consider adding tax rate (percentage value)  to line item data? Now we have the the tax_class, total_tax etc. But to get the rate we need to either count it or get from the Tax Rate endpoint. When having a lot of order data this makes processes a bit slower or adds traffic.

**Comment by Laserology** [comment]
Specific modular "implementations" of sorts could work well. In California, you're only required to report the cities you made sales in, and how much in those cities. No rates or percentages of those values. This is vastly different from the usual way that woo-commerce tax works and having different "modules" that users can choose to install could help with making the varying needs easier to cover.

Large businesses established in California still have to report taxes for other states or regions normally if they have a significant enough presence aswell in any other states. But any given business may only sell to certain areas, so allowing the user to only use the modules that they need would simplify taxes greatly.

Hope my input is useful - Thanks for all the hard work you guys do on this software.
